### ISSUE TITLE
Clarification on LiDAR RGB field and intensity handling in the HRDX point clouds

### ISSUE THREAD

**Issue by guyupan1911** [honda-research-institute/HRDX#2]
Hi, thank you for releasing the HRDX dataset and code.

I have a question regarding the fourth feature of the LiDAR point clouds.

For example, the following HRDX PCD file:

pointclouds/2025_05_01_12_54_30/286_1746129302_220.pcd

has the following header:

FIELDS x y z rgb
SIZE 4 4 4 4
TYPE F F F F
COUNT 1 1 1 1

Therefore, the fourth field appears to be a PCL-style packed RGB value rather than an explicit LiDAR intensity or reflectivity value.

However, in plugin/datasets/pipelines/loading_lidar.py, the RGB field is handled as follows:

# Re-interpret `rgb` packed float as raw bytes -> single channel intensity proxy
# (lowest byte of the packed RGB int).
if fname == 'rgb' and col.dtype.kind == 'f':
    col = col.view(np.uint32).astype(np.float32)

It seems that this implementation converts the complete packed 0xRRGGBB value to float32 rather than extracting the lowest byte as stated in the comment. The resulting value can be as large as approximately 16.7 million.

The fourth column is then treated as intensity by LidarPillarEncoder:

has_intensity = pts_sel.shape[1] >= 4
i = pts_sel[:, 3] if has_intensity else torch.zeros_like(x)

I also noticed that the RGB values look like a pseudocolor palette similar to JET rather than natural camera colors.

Could you please clarify the following questions?

1. What does the rgb field in the HRDX PCD files represent?
2. Was the original LiDAR intensity or reflectivity mapped to RGB using a colormap?
3. If so, which colormap, normalization range, clipping rule, or nonlinear transformation was used?
4. Is it possible to recover the original intensity values from the packed RGB field?
5. Is the intended loader behavior to use the full packed 0xRRGGBB integer, the lowest byte, a grayscale value, or an inverse-colormap intensity proxy?
6. Are the original XYZI point clouds available?

At the moment, the HRDX LiDAR pipeline effectively passes:

[x, y, z, packed_rgb_integer]

to a module that expects:

[x, y, z, intensity]

Understanding the intended representation would help ensure that the LiDAR fusion experiments use the correct point features.

Thank you!
